builtin/sampled_dataset.cc

```diff
@@ -0,0 +1,368 @@
+/** sampled_dataset.cc                                              -*- C++ -*-
+    Francois Maillet, 11 janvier 2016
+    This file is part of MLDB. Copyright 2015 Datacratic. All rights reserved.
+
+*/
+
+#include "mldb/sql/sql_expression.h"
+#include "sampled_dataset.h"
+#include "mldb/jml/utils/lightweight_hash.h"
+#include "mldb/types/any_impl.h"
+#include "mldb/types/structure_description.h"
+#include <random>
+#include <unordered_set>
+
+using namespace std;
+
+
+namespace Datacratic {
+namespace MLDB {
+
+    
+/*****************************************************************************/
+/* SAMPLED DATASET CONFIG                                                    */
+/*****************************************************************************/
+
+std::random_device rd;
+
+SampledDatasetConfig::
+SampledDatasetConfig() :
+        rows(0), fraction(0), withReplacement(false)
+{
+    seed = rd();
+}
+
+
+/*****************************************************************************/
+/* SAMPLED DATASET CONFIG                                                 */
+/*****************************************************************************/
+
+DEFINE_STRUCTURE_DESCRIPTION(SampledDatasetConfig);
+
+SampledDatasetConfigDescription::
+SampledDatasetConfigDescription()
+{
+    nullAccepted = true;
+
+    addField("rows", &SampledDatasetConfig::rows,
```

Comment:
I think rows makes more sense: we are sampling either n rows OR f fraction (see next option)